crates/fs/private/namefilter/bloomfilter.rs

```diff
@@ -0,0 +1,172 @@
+use std::ops::Index;
+
+use bitvec::prelude::BitArray;
+use xxhash_rust::xxh3;
+
+//------------------------------------------------------------------------------
+// Type Definitions
+//------------------------------------------------------------------------------
+
+/// The bloom filter is a probabilistic data structure that can be used to store a set of hashes.
+///
+/// `N` is the size of the bloom filter in bytes.
```

Comment:
```suggestion
/// `N` is the size of the bloom filter in bytes. Must be a power of two.
```

I'm thinking we should maybe just throw that in there, since we're doing a `% (N * 8)` later on, that is supposed to be uniformly distributed.

We could *maybe* also make the parameter `N` specify the power of bits? Like `BitArray<[u8; 1 << N]>`. Dunno. Let me know what you think, this may be too many indirections until you know the actual bit size.

Reply:
Ugh, whatever, it's somewhat premature abstraction anyway, we're not using different bloom sizes. So it doesn't really matter that much.